plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/ComparePresenter.java

```diff
@@ -0,0 +1,111 @@
+import org.eclipse.che.ide.rest.StringUnmarshaller;
+
+/**
+ * Presenter for compearing current files with files from specified working tree
+ * 
+ * @author Igor Vinokur
+ */
+@Singleton
+public class ComparePresenter implements CompareView.ActionDelegate {
+
+    private final AppContext           appContext;
+    private final CompareView          view;
+    private final ProjectServiceClient projectService;
+    private final GitServiceClient     gitService;
+    private final NotificationManager  notificationManager;
+
+    @Inject
+    public ComparePresenter(AppContext appContext,
+                            CompareView view,
+                            ProjectServiceClient projectService,
+                            GitServiceClient gitServiceClient,
+                            NotificationManager notificationManager) {
+        this.appContext = appContext;
+        this.view = view;
+        this.projectService = projectService;
+        this.gitService = gitServiceClient;
+        this.notificationManager = notificationManager;
+        this.view.setDelegate(this);
+    }
+
+    /**
+     * Show dialog.
+     */
+    public void show(final String file, final String state, final String revision) {        
+        if (state.contains("A")) {
+            showCompare(file, "", revision);
+        } else if (state.contains("D")) {
+            gitService.show(appContext.getCurrentProject().getRootProject(), file, revision,
+                        new AsyncRequestCallback<String>(new StringUnmarshaller()) {
+                            @Override
+                            protected void onSuccess(final String oldContent) {
+                                view.setTitle(file);
+                                view.show(oldContent, "", revision, file.substring(file.lastIndexOf("/")));
+                            }
+
+                            @Override
+                            protected void onFailure(Throwable exception) {
+                                notificationManager.showError(exception.getMessage());
+                            }
+                        });
+        } else {
+            gitService.show(appContext.getCurrentProject().getRootProject(), file, revision,
+                        new AsyncRequestCallback<String>(new StringUnmarshaller()) {
+                            @Override
+                            protected void onSuccess(final String oldContent) {
+                                showCompare(file, oldContent, revision);
+                            }
+
+                            @Override
+                            protected void onFailure(Throwable exception) {
+                                notificationManager.showError(exception.getMessage());
+                            }
+                        });
+        }    
+        
+    }
+
+    private void showCompare(final String item, final String oldContent, final String revision) {
```

Comment:
Well, if you don't mind could you please explain your `actually it doesn't matter :)` ?